### PR TITLE
Implement CC bans

### LINKED ISSUE #67: Implement banning users from CC module

### PR DESCRIPTION
**Pull request by GTG3000**

### CODE REVIEW

File: plugins/custom_commands.py [hunk 1]
```diff
@@ -239,6 +260,51 @@ class CustomCommands(BasePlugin):
...
+
+        if t_member.id in self.plugin_config[gid]["cc_use_ban"]:
+            self.plugin_config[gid]["cc_use_ban"].remove(t_member.id)
+            await respond(msg, f"**AFFIRMATIVE. User {t_member.mention} was allowed the usage of custom commands.**")
+        else:
+            self.plugin_config[gid]["cc_use_ban"].append(t_member.id)
+            await respond(msg, f"**AFFIRMATIVE. User {t_member.mention} was banned from using custom commands.**")
+
+    @Command("ccban", "bancc",
+             doc="Toggles users ability to crate and alter custom commands.",
```
Comment (medeor413) [inline plugins/custom_commands.py]: create*

File: plugins/custom_commands.py [hunk 2]
```diff
@@ -115,6 +126,12 @@ class CustomCommands(BasePlugin):
              syntax="(name) (content)",
              category="custom_commands")
     async def _editcc(self, msg):
+        gid = str(msg.guild.id)
+        if gid not in self.plugin_config:
+            self.plugin_config[gid] = DotDict(self.default_config["default"])
+            self.config_manager.save_config()
+        if msg.author.id in self.plugin_config[gid]["cc_create_ban"]:
+            raise PermissionError("Banned from editing custom commands.")
```
Comment (medeor413) [inline plugins/custom_commands.py]: You are banned*

File: plugins/custom_commands.py [hunk 3]
```diff
@@ -145,11 +161,16 @@ class CustomCommands(BasePlugin):
              syntax="(name)",
              category="custom_commands")
     async def _delcc(self, msg):
+        gid = str(msg.guild.id)
+        if gid not in self.plugin_config:
+            self.plugin_config[gid] = DotDict(self.default_config["default"])
+            self.config_manager.save_config()
+        if msg.author.id in self.plugin_config[gid]["cc_create_ban"]:
+            raise PermissionError("Banned from deleting custom commands.")
```
Comment (medeor413) [inline plugins/custom_commands.py]: You are banned*

File: plugins/custom_commands.py [hunk 4]
```diff
@@ -239,6 +260,51 @@ class CustomCommands(BasePlugin):
...
+             syntax="(user)",
+             category="custom_commands",
+             perms={"manage_messages"})
+    async def _mutecc(self, msg):
+        gid = str(msg.guild.id)
+        if gid not in self.plugin_config:
+            self.plugin_config[gid] = DotDict(self.default_config["default"])
+            self.config_manager.save_config()
+
+        args = msg.content.split(" ", 1)
```
Comment (medeor413) [inline plugins/custom_commands.py]: shlex.split

File: plugins/custom_commands.py [hunk 5]
```diff
@@ -239,6 +260,51 @@ class CustomCommands(BasePlugin):
...
+             syntax="(user)",
+             category="custom_commands",
+             perms={"manage_messages"})
+    async def _bancc(self, msg):
+        gid = str(msg.guild.id)
+        if gid not in self.plugin_config:
+            self.plugin_config[gid] = DotDict(self.default_config["default"])
+            self.config_manager.save_config()
+
+        args = msg.content.split(" ", 1)
```
Comment (medeor413) [inline plugins/custom_commands.py]: shlex.split

File: plugins/custom_commands.py [hunk 6]
```diff
@@ -14,7 +14,9 @@ class CustomCommands(BasePlugin):
     default_config = {
         "cc_file": "config/ccs.json",
         "default": {
-            "cc_prefix": "!!"
+            "cc_prefix": "!!",
+            "cc_create_ban": [],
```
Comment (medeor413) [inline plugins/custom_commands.py]: Storing this in config is *okay*, but the shelve storage is more suitable for this sort of thing.

File: plugins/custom_commands.py [hunk 7]
```diff
@@ -64,6 +66,10 @@ class CustomCommands(BasePlugin):
         if gid not in self.plugin_config:
             self.plugin_config[gid] = DotDict(self.default_config["default"])
             self.config_manager.save_config()
+        if msg.author.id in self.plugin_config[gid]["cc_use_ban"]:
+            await msg.author.send(f"**WARNING: You are banned from usage of custom command on the server "
```
Comment (medeor413) [inline plugins/custom_commands.py]: commands*

File: plugins/custom_commands.py [hunk 8]
```diff
@@ -83,6 +89,11 @@ class CustomCommands(BasePlugin):
              category="custom_commands")
     async def _createcc(self, msg):
         gid = str(msg.guild.id)
+        if gid not in self.plugin_config:
+            self.plugin_config[gid] = DotDict(self.default_config["default"])
+            self.config_manager.save_config()
+        if msg.author.id in self.plugin_config[gid]["cc_create_ban"]:
+            raise PermissionError("Banned from creating custom commands.")
```
Comment (medeor413) [inline plugins/custom_commands.py]: You are banned*

File: plugins/custom_commands.py (hunk 4 above)
Comment (GTG3000) [reply plugins/custom_commands.py]: is an overkill

File: plugins/custom_commands.py (hunk 5 above)
Comment (GTG3000) [reply plugins/custom_commands.py]: is an overkill

Comment (medeor413) [reply plugins/custom_commands.py]: names can have spaces

**Pull request merged**